Dockerfile

```diff
@@ -30,30 +30,44 @@ RUN yum -y --setopt=tsflags=nodocs update && \
            /var/opennms/reports && \
     mkdir -p /opennms-data/logs \
              /opennms-data/rrd \
-             /opennms-data/reports && \
+             /opennms-data/mibs \
+             /opennms-data/reports \
+             /opt/opennms-etc-overlay \
+             /opt/opennms/assets && \
+    mv /var/opennms/mibs/compiled /opennms-data/mibs/ && \
+    rm -rf /var/opennms/mibs && \
     ln -s /opennms-data/logs /opt/opennms/logs && \
     ln -s /opennms-data/rrd /var/opennms/rrd && \
-    ln -s /opennms-data/reports /var/opennms/reports
+    ln -s /opennms-data/mibs /var/opennms/mibs && \
+    ln -s /opennms-data/reports /var/opennms/reports && \
+    mkdir -p /var/opennms/data-pristine && \
+    cp -r /opennms-data/* /var/opennms/data-pristine/ && \
+    sed -r -i '/RUNAS/s/root/opennms/' /opt/opennms/bin/opennms && \
+    sed -r -i '/RUNAS/s/root/opennms/' /opt/opennms/bin/install && \
+    sed -r -i '/RUNAS/s/root/opennms/' /opt/opennms/bin/upgrade && \
+    sed -r -i 's/"162"/"1162"/' /opt/opennms/etc/trapd-configuration.xml && \
+    groupadd -r opennms && useradd -r -g opennms -d /opt/opennms opennms && \
+    chown opennms:opennms -R /opt/opennms /opennms-data /opt/opennms-etc-overlay && \
+    chgrp -R 0 /opt/opennms /opennms-data /opt/opennms-etc-overlay && \
+    chmod -R g=u /opt/opennms /opennms-data /opt/opennms-etc-overlay
 
-COPY ./assets/opennms-datasources.xml.tpl /root
-COPY ./assets/org.apache.karaf.shell.cfg.tpl /root
-COPY ./assets/newts.properties.tpl /root
+COPY ./assets/opennms-datasources.xml.tpl /opt/opennms/assets
+COPY ./assets/org.apache.karaf.shell.cfg.tpl /opt/opennms/assets
+COPY ./assets/newts.properties.tpl /opt/opennms/assets
 
 COPY ./docker-entrypoint.sh /
 
-## Volumes for storing data outside of the container
-VOLUME [ "/opt/opennms/etc", "/opt/opennms-etc-overlay", "/opennms-data" ]
-
 LABEL license="AGPLv3" \
       org.opennms.horizon.version="${OPENNMS_VERSION}" \
       vendor="OpenNMS Community" \
       name="Horizon"
 
 WORKDIR /opt/opennms
+USER 999
```

Comment:
Just because curiosity, the [example](https://github.com/RHsyseng/container-rhel-examples/blob/master/starter-arbitrary-uid/Dockerfile) of their image use the ID 10001? Do you know why?

Reply:
OpenShift uses high numbers for the random UID you're going to get at runtime, maybe that's why they use that ID. In our case, we should use whatever numbered ID we got when the "opennms" user is added through the `useradd` command. Of course, we could pass the desired UID we would like to use, for consistency. I believe it worth doing that, here is why:
```
[root@39815013d19e opennms]# useradd opennms
[root@39815013d19e opennms]# grep opennms /etc/group
opennms:x:1000:
[root@39815013d19e opennms]# grep opennms /etc/passwd
opennms:x:1000:1000::/home/opennms:/bin/bash
```
So, this time, I got `1000` instead of `999`.